src/math/bigintops.c

```diff
@@ -359,6 +359,11 @@ MVMObject * MVM_bigint_##opname(MVMThreadContext *tc, MVMObject *result_type, MV
         mp_##opname(ia, ib, ic); \
         store_bigint_result(bc, ic); \
         clear_temp_bigints(tmp, 2); \
+        if (mp_count_bits(ic) > 65 && \
+            ((tc->nursery_alloc_limit - sizeof(mp_int)) > (tc->nursery_alloc + sizeof(mp_int)))) \
+        { \
+            tc->nursery_alloc_limit -= sizeof(mp_int); \
```

Comment:
This only adds a constant amount of pressure, rather than being proportional to the size of the bigint itself (which `mp_count_bits` can advise on since we already call it - however, I suspect there's a faster way, like just reading the `->used` or whatever it's called of the `mp_int`, which will give a number in bytes). I'd suggest a scheme like:

* Apply memory pressure equivalent to the size of the bigint storage, and mask the lower three bits so we don't lose 8-byte alignment (this freely handles bigints that use less than 8 bytes not causing any added pressure, which seems fair enough). Note the loss of alignment can cause crashes on some platforms, though on x86/x64 would just result in a slowdown.
* Put put an upper limit on the pressure of something like 32KB so we won't just GC madly on such values